mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/MaildirFolder.java

```diff
@@ -339,9 +341,9 @@ private void saveMailboxId(MaildirId id) throws IOException {
      * Sets and returns a new uidValidity for this folder.
      * @return the new uidValidity
      */
-    private long resetUidValidity() throws IOException {
+    private UidValidity resetUidValidity() throws IOException {
         // using the timestamp as uidValidity
-        long timestamp = System.currentTimeMillis();
+        UidValidity timestamp = UidValidity.of(System.currentTimeMillis());
```

Comment:
```
jshell> System.currentTimeMillis()/ Integer.MAX_VALUE
$2 ==> 736
```
That means this piece of code doesn't respect UIDVALIDITY definition.

Reply:
BTW here I believe this code path SHOULD be using `UidValidity.random()`, I don't see reasons to be relying on system clock.

Reply:
Because it's what RFC 3501 advise.
However, not with millis because it doesn't fit in unsigned 32 bit integer.

Reply:
How can you ensure 2 mailboxes don't get the same value if they are created at the exact same time in two different threads?

Reply:
It doesn't matter to have 2 differents mailboxes having the same uidvalidity

Reply:
Not if you "exchange" mbx names

A-> c
B-> a
C -> b

Reply:
I don't get it

Reply:
Imagine... I create mailboxA and mailboxB at the same time. They have same UID_VALIDITY, let say 36.

Now I fetch my mails, then shut thunderbird down.

Then I rename mailboxA in mailboxC
Rename mailboxB in mailboxA
Rename mailboxC in mailboxB

Now I turn thunderbird on again. Fetch the mailboxes again and sees mailboxA with UIDVALIDITY 36 + mailboxB with UID validity 36 and don't know it SHOULD drop it's cache, and resynchronise...

Yet another example about IMAP RFCs mess IMO. That's why I think keeping this in our code, even if it is an IMAP RFC advice is not wise.

Reply:
Oh, good scenario, it demonstrates UIDVALIDITY is part of mailbox identity and as mailbox path is a weak identifier, you suggest we use UIDVALIDITY as a unique id.

I agree with that solution.

BTW, I already pointed out that using millisecond in UIDVALIDITY is not working neither because it's 32 bits.

Reply:
Should we keep a record of this discussion somewhere @mbaechler ? ADR? Or link in the ticket?

Reply:
I would vote for a comment in the code